src/plugins/wizard/public/plugin.ts

```diff
@@ -92,6 +93,36 @@ export class WizardPlugin
         // Instantiate the store
         const store = await getPreloadedStore(services);
 
+        let previousStore = {
+          vizState: store.getState().visualization,
+          styleState: store.getState().style,
+        };
+
+        let previousMetadata = store.getState().metadata;
+
+        // Listen to changes
+        const handleChangee = () => {
+          const currentStore = {
+            vizState: store.getState().visualization,
+            styleState: store.getState().style,
+          };
+
+          const metaState = store.getState().metadata;
+
+          if (
+            metaState.editorState.state === 'clean' &&
+            JSON.stringify(currentStore) !== JSON.stringify(previousStore) &&
+            previousMetadata.editorState.state === 'clean'
+          ) {
+            store.dispatch(setSaveState({ state: 'dirty' }));
+          }
+
+          previousStore = currentStore;
+          previousMetadata = metaState;
+        };
+
+        const unsubscribe = store.subscribe(handleChangee);
```

Comment:
can we move all this logic into `getPreloadedStore`?  will keep `plugin.ts` more readable, and nothing that we do here is needed for `renderApp`